src/Resources/Resources/Models.Authorization/AuthorizationClient.cs

```diff
@@ -187,7 +187,20 @@ public List<PSRoleAssignment> FilterRoleAssignments(FilterRoleAssignmentsOptions
             string principalId = null;
             PSADObject adObject = null;
             ODataQuery<RoleAssignmentFilter> odataQuery = null;
-
+            // refer to https://learn.microsoft.com/en-us/azure/role-based-access-control/role-assignments-list-rest
+            // https://learn.microsoft.com/en-us/azure/role-based-access-control/elevate-access-global-admin#elevate-access-for-a-global-administrator-1
+            // scope is path variable in REST API. When scope is '/', $filter=atScope() is required, or else it will throw BadRequest.
+            // $filter=atScope() to return all role assignments at or above the scope. or else it returns all role assignments at, above or below the scope for the specific scope.
+            //
+            // Known issue:
+            // Before change: This command will return all role assignments at, above or below the scope for the specific scope.
+            // After change: This command will return all role assignments at the scope for the specific scope.
+            // But Azure cli az role assignment list --scope will only return role assignments at the specific scope, --include-inherited will return the assignments above the scope
+            // In order to implement this feature, it needs to add a new parameter IncludeInherited.
```

Comment:
This block of comments is very hard to read.
Do you intend to return role assignments at scope for ALL scopes or only if no scope is provided? (i.e. intended scope is "/")

Reply:
> This block of comments is very hard to read. Do you intend to return role assignments at scope for ALL scopes or only if no scope is provided? (i.e. intended scope is "/")

With -scope set in the command, the behavior in powershell is different from cli command.  If you check the document, this is a bug. If you think the change is too much, I can change it to check scope '/' only, but please be aware of this existing bug.
https://learn.microsoft.com/en-us/powershell/module/az.resources/get-azroleassignment?view=azps-9.1.0#-scope

Reply:
I have removed the extra fix for -scope, only check '/' to fix BadRequest here.

Reply:
I have taken note of the bug and will address after we migrate to autogeneration.
Please do not change the behavior in this PR